### PR TITLE
docs(books): solved two issues

### PR DESCRIPTION
**Pull request by LuigiImVector**
## What does this PR do?
Add resource(s)

### Description
Added two books suggested in #6656 and #6685 (maths and game development in C).

### Why is this valuable (or not)?
Both are serious projects that can be useful to the comunity.

### How do we know it's really free?

3D Math Primer for Graphics and Game Development is published in the author's official website, he share it.
Mathematics for Computer Science has [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0/) license.

### For book lists, is it a book? For course lists, is it a course? etc.
Books.

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [X] Search for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -421,6 +421,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 ### C
 
 * [256-Color VGA Programming in C](http://www.brackeen.com/vga/) - David Brackeen
+* [3D Math Primer for Graphics and Game Development](https://gamemath.com/book/intro.html) - Fletcher Dunn (HTML)
```
Comment (LuigiImVector) [inline books/free-programming-books-langs.md:424]: Is it in the most appropriate section? "Game Development" category in `free-programming-books-subjects` is better?

Comment (eshellman) [reply books/free-programming-books-langs.md:424]: I'd go with game dev

Comment (LuigiImVector) [reply books/free-programming-books-langs.md]: Done

**Comment by eshellman** [comment]
thanks for seeing after the issue list

**Comment by LuigiImVector** [comment]
Sorry but I didn't quite understand why it was closed @eshellman

**Comment by eshellman** [comment]
mistake

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -262,6 +262,7 @@ Books that cover a specific programming language can be found in the  [BY PROGRA
 ### Game Development
 
 * [2D Game Development: From Zero To Hero](https://github.com/Penaz91/2DGD_F0TH) - Daniele Penazzo (HTML, [PDF, EBPUB, Kindle...](https://therealpenaz91.itch.io/2dgd-f0th#download)) (:construction: *in process*)
+* [3D Math Primer for Graphics and Game Development](https://gamemath.com/book/intro.html) - Fletcher Dunn (HTML)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:265]: Should this use top url?

https://gamemath.com/book

Comment (eshellman) [reply books/free-programming-books-subjects.md:265]: no

**Pull request merged**